packages/utilities/src/styled.tsx

```diff
@@ -105,7 +105,9 @@ export function styled<
 
         return () => Customizations.unobserve(forceUpdate);
       }
-    }, []);
+      // TODO: check with @MLoughry about whether to add dep on
```

Comment:
@MLoughry would appreciate your input on this

Reply:
It shouldn't have any deps, as it should only run on mount/unmount. Deps in `useEffect` control *when* it's run, and thus needn't be exhaustive. Is the rule throwing in this case?

Reply:
The rule flags *any* time you have a non-exhaustive deps list, because that often (but not always) points to an error in the logic.

Reply:
I changed it to disable the rule in this case. The reason I asked though instead of just disabling the rule to begin with was because I wondered if it would actually be appropriate to re-configure if `inCustomizerContext` changed (which I'm not sure if it can due to not being familiar with Customizer).

Reply:
I don't think it's possible. It's set to true if a parent component is a `<CustomizerContext>`, and I don't think a React component instance can be re-homed under a separate tree.